### PR TITLE
Enhance pogo cpp templates

### PR DESCRIPTION
**Pull request by t-b**

### CODE REVIEW

File: assets/pogo/templates/cpp/Allowed.cpp [hunk 1]
```diff
@@ -1,22 +1,12 @@
-static const char *RcsId = "$Header$";
 //+=============================================================================
 //
 // file :         TemplateDevServ.cpp
 //
-// description :  C++ source for the TemplateDevServ and its alowed. 
+// description :  C++ source for the TemplateDevServ and its alowed.
```
Comment (bourtemb) [inline assets/pogo/templates/cpp/Allowed.cpp]: Let's take this opportunity to fix some typos:
"alowed." => "allowed"

File: assets/pogo/templates/cpp/Allowed.cpp [hunk 2]
```diff
@@ -1,22 +1,12 @@
...
 //                method for commands and attributes
```
Comment (bourtemb) [inline assets/pogo/templates/cpp/Allowed.cpp]: method => methods

File: assets/pogo/templates/cpp/MachineState.cpp [hunk 3]
```diff
@@ -1,34 +1,12 @@
-static const char *RcsId = "$Id $";
 //+=============================================================================
 //
 // file :         TemplateDevServStateMachine.cpp
 //
-// description :  C++ source for the TemplateDevServ and its alowed 
+// description :  C++ source for the TemplateDevServ and its alowed
```
Comment (bourtemb) [inline assets/pogo/templates/cpp/MachineState.cpp]: alowed => allowed

**Review by bourtemb — COMMENTED** [review]
Thanks @t-b for this useful Pull Request.
Let's take this opportunity to fix a few typos.

**Comment by t-b** [comment]
@bourtemb Done.

**Review by bourtemb — CHANGES_REQUESTED** [review]
I tested the changes and tried to generate a device server source code using a POGO installed from this branch of the SourceDistribution.
I noticed that this new version of the templates didn't seem to be taken into account by POGO (still no trace of std::string in the generated code).
So I investigated a bit further and I discovered that these templates do not seem to be used any longer by POGO?
It looks like the templates which are used are the ones in https://github.com/tango-controls/pogo/tree/master/fr.esrf.tango.generator.xtend/src/fr/esrf/tango/pogo/generator/cpp
@Pascal-Verdier , could you confirm, please?

If it is confirmed these POGO templates from the source distribution are useless, we should remove them and update the POGO startup script which seems to define some variables to point to these templates.

We should also update the files in https://github.com/tango-controls/pogo/tree/master/fr.esrf.tango.generator.xtend/src/fr/esrf/tango/pogo/generator/cpp in the POGO git repository.

**Comment by bourtemb** [comment]
I got confirmation from @Pascal-Verdier that these templates provided with the source distribution are now obsolete. They were used by Pogo 6. We should remove them from the Source distribution.
Now, the ones which are used are the ones in https://github.com/tango-controls/pogo/tree/master/fr.esrf.tango.generator.xtend/src/fr/esrf/tango/pogo/generator/cpp
So, a similar Pull Request should be proposed in [POGO Github repository](https://github.com/tango-controls/pogo).

**Comment by t-b** [comment]
@bourtemb So the correct ones are https://github.com/tango-controls/pogo/tree/master/fr.esrf.tango.generator.xtend/src/fr/esrf/tango/pogo/generator/cpp?

**Comment by bourtemb** [comment]
> 
> 
> @bourtemb So the correct ones are https://github.com/tango-controls/pogo/tree/master/fr.esrf.tango.generator.xtend/src/fr/esrf/tango/pogo/generator/cpp?

Yes

**Comment by t-b** [comment]
@bourtemb Okay I'll create a new PR on the pogo repo. Can you then delete these old files here?

**Comment by bourtemb** [comment]
> @bourtemb Okay I'll create a new PR on the pogo repo. Can you then delete these old files here?

@t-b, thanks! Sure, I'll take care of removing these old files here.

**Pull request closed without merging**